pkg/cluster/sync.go

```diff
@@ -1481,3 +1646,223 @@ func (c *Cluster) createMonitoringSecret() error {
 
 	return nil
 }
+
+func generateRootCertificate(
```

Comment:
We will need to optionally integrate with cert-manager to have it provide certificates for us. Probably at some point this needs to be moved behind an interface. Fine to have it hard coded initially.